pkg/preparation/sqlrepo/stats.go

```diff
@@ -0,0 +1,101 @@
+package sqlrepo
+
+import (
+	"context"
+
+	"github.com/storacha/guppy/pkg/preparation/types/id"
+)
+
+func (r *Repo) TotalBytesToScan(ctx context.Context, uploadID id.UploadID) (uint64, error) {
+	row := r.db.QueryRowContext(
+		ctx, `
+			SELECT COALESCE(SUM(fs_entries.size), 0)
```

Comment:
mild anxiety about the perf on this query at scale -- we could have maybe 100,000+ files in some uploads